src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -32,6 +32,8 @@ public function __construct(array $decoders = array())
 
     /**
      * {@inheritdoc}
+     * 
+     * @throws RuntimeException If no decoder is found.
```

Comment:
>If the interface does not describe in which case it is throw, the phpdoc is useless as calling code cannot handle it anyway (it cannot know which kind of error case will throw it, and so cannot decide how to recover).

I don't agree with that: you may handle differently an exception coming from a serialization error than a generic one.